src/components/common/DeleteConfirmationModal.test.tsx

```diff
@@ -0,0 +1,139 @@
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { DeleteConfirmationModal } from 'src/components/common/DeleteConfirmationModal';
+import { renderWithAppContexts } from 'src/testing/test-utils';
+
+describe('DeleteConfirmationModal', () => {
+  it('renders a confirmation modal dialog with the type/name passed in', () => {
+    // Act
+    renderWithAppContexts(
+      <DeleteConfirmationModal objectType='nut' objectName='pistachio' onConfirm={jest.fn()} onDismiss={jest.fn()} />
+    );
+
+    // Assert
+    const modalElement = screen.getByRole('dialog');
+    expect(modalElement).toHaveAccessibleName('Delete nut');
+    expect(modalElement).toHaveTextContent(/Are you sure you want to delete the nut pistachio?/);
+  });
+
+  it('allows the specification of a custom title', () => {
+    // Act
+    renderWithAppContexts(
+      <DeleteConfirmationModal
+        title='Allergic to nuts?'
+        objectType='nut'
+        objectName='pistachio'
+        onConfirm={jest.fn()}
+        onDismiss={jest.fn()}
+      />
+    );
+
+    // Assert
+    const modalElement = screen.getByRole('dialog');
+    expect(modalElement).toHaveAccessibleName('Allergic to nuts?');
+  });
+
+  it('allows passing custom children', () => {
+    // Act
+    renderWithAppContexts(
+      <DeleteConfirmationModal
+        title='Allergic to nuts?'
+        objectType='nut'
+        objectName='pistachio'
+        onConfirm={jest.fn()}
+        onDismiss={jest.fn()}
+      >
+        <div>custom message</div>
+      </DeleteConfirmationModal>
+    );
+
+    // Assert
+    const modalElement = screen.getByRole('dialog');
+    expect(modalElement).toHaveTextContent(/custom message/);
+    expect(modalElement).not.toHaveTextContent(/Are you sure you want to delete the nut pistachio?/);
+  });
+
+  it('supports forcing the user to type confirmation text (case insensitive)', async () => {
+    // Arrange
+    const user = userEvent.setup();
+    const isConfirmButtonDisabled = () => {
+      const confirmButton = screen.getAllByRole('button').find((button) => button.textContent === 'Delete nut');
+      return confirmButton!.getAttribute('aria-disabled');
+    };
+
+    // Act
+    renderWithAppContexts(
+      <DeleteConfirmationModal
+        confirmationPrompt='confirm'
+        objectType='nut'
+        objectName='pistachio'
+        onConfirm={jest.fn()}
+        onDismiss={jest.fn()}
+      />
+    );
+    expect(isConfirmButtonDisabled()).toBe('true');
+    const confirmationInput = screen.getByLabelText('Type "confirm" to continue:');
+    await user.type(confirmationInput, 'conFIRM');
```

Comment:
I don't think there are any uses of this in the product (the "Delete Workspace" modal, which requires confirmation, is not using this shared modal). However, I hacked the code to test it out locally:

![image](https://github.com/user-attachments/assets/5a48793a-7483-48cc-a484-fea1a54a1b08)

Reply:
This may be the modal a user gets when deleting a group.